specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json

```diff
@@ -667,25 +675,124 @@
           }
         },
         "selectorMatchOperator": {
-          "type": "string",
-          "enum": [
-            "Equals",
-            "Contains",
-            "StartsWith",
-            "EndsWith",
-            "EqualsAny"
-          ],
           "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to.",
-          "x-ms-enum": {
-            "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
-            "modelAsString": true
-          }
+          "$ref": "#/definitions/MatchOperator"
         },
         "selector": {
           "type": "string",
           "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
         }
       }
+    },
+    "MatchOperator": {
+      "type": "string",
+      "enum": [
+        "Equals",
+        "Contains",
+        "StartsWith",
+        "EndsWith",
+        "EqualsAny"
+      ],
+      "description": "Defines the logic for comparing two strings.",
+      "x-ms-enum": {
+        "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
```

Comment:
Is it only for OwaspCrsExclusions still? I think you are now using this for two different purposes right? What will the consequences be for existing uses now that we are changing from inline to a ref?

Reply:
Not sure about the consequences
But you bring up a good point. I think I'll keep them separate in that case, since they can potentially diverge in the future